src/Symfony/Component/Validator/Constraints/TimezoneValidator.php

```diff
@@ -0,0 +1,98 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Validator\Constraints;
+
+use Symfony\Component\Validator\Constraint;
+use Symfony\Component\Validator\ConstraintValidator;
+use Symfony\Component\Validator\Exception\UnexpectedTypeException;
+
+/**
+ * Validates whether a value is a valid timezone identifier.
+ *
+ * @author Javier Spagnoletti <phansys@gmail.com>
+ */
+class TimezoneValidator extends ConstraintValidator
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function validate($value, Constraint $constraint)
+    {
+        if (!$constraint instanceof Timezone) {
+            throw new UnexpectedTypeException($constraint, __NAMESPACE__.'\Timezone');
+        }
+
+        if (null === $value || '' === $value) {
+            return;
+        }
+
+        if (!is_scalar($value) && !(is_object($value) && method_exists($value, '__toString'))) {
+            throw new UnexpectedTypeException($value, 'string');
+        }
+
+        $value = (string) $value;
+        $zone = null !== $constraint->zone ? $constraint->zone : \DateTimeZone::ALL;
+        $timezoneIds = \DateTimeZone::listIdentifiers($zone, $constraint->countryCode);
+
+        if ($timezoneIds && !in_array($value, $timezoneIds, true)) {
+            if ($constraint->countryCode) {
+                $code = Timezone::NO_SUCH_TIMEZONE_IN_COUNTRY_ERROR;
+            } elseif (null !== $constraint->zone) {
+                $code = Timezone::NO_SUCH_TIMEZONE_IN_ZONE_ERROR;
+            } else {
+                $code = Timezone::NO_SUCH_TIMEZONE_ERROR;
+            }
+
+            $this->context->buildViolation($constraint->message)
+                ->setParameter('{{ extra_info }}', $this->formatExtraInfo($constraint->zone, $constraint->countryCode))
```

Comment:
This is a weird parameter, and binds localized text to this constraint validator... :-1: for that.

What about specialized params instead like `zone_id` / `zone_name` or so?